src/core/Typography/Text/Text.tsx

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { useTheme } from '../../utils/hooks/useTheme';
+import { CommonProps } from '../../utils/props';
+import '@itwin/itwinui-css/css/text.css';
+
+type TextOwnProps<T extends React.ElementType | React.ComponentType = 'div'> = {
+  /**
+   * What element should the text be rendered as?
+   *
+   * @default 'div'
+   */
+  as?: T;
+  /**
+   * Which typography variant/size should be used for the styling?
+   *
+   * 'headline' = h1,
+   * 'title' = h2,
+   * 'subheading' = h3,
+   * 'leading' = h4,
+   * 'body' = normal paragraph text,
+   * 'small' = smaller text
+   *
+   * @default 'body'
+   */
+  variant?: 'headline' | 'title' | 'subheading' | 'leading' | 'body' | 'small';
+  /**
+   * Show text in muted style.
+   * @default false
+   */
+  isMuted?: boolean;
+  /**
+   * Use it if you are still loading the content.
+   * @default false
+   */
+  isSkeleton?: boolean;
+
+  /**
+   * Does the element have a bottom margin?
+   * @default false
+   */
+  hasMargin?: boolean;
```

Comment:
I am not sure about this prop. Do we really need it?

Reply:
I think the ultimate goal would to replace all Typography components with this. Since many of them have margin, it should be possible to achieve the same here.